### ISSUE TITLE
Please, use modern Nix submodule

### ISSUE THREAD

**Issue by Anton-Latukha** [haskell-nix/hnix#610]
_(#966 is required before this one.)_

The submodule is frozen for quite some time (rev. Oct 10, 2019).

The project needs to follow or at least use more recent Nix release.

Actually, old module checkout creates an incompatibility with building the project itself in Nixpkgs ecosystem.

On Nixpkgs `HNix 0.8.0` build, since the project relies on old module here, and so - HNix does not passes its own testing suite in Nixpkgs using newer Nix:

<details><summary>Tail of the build log</summary>
<p>

```
running tests
Running 1 test suites...
Test suite hnix-tests: RUNNING...
hnix
  ParserTests
    constant int:                                                    OK
    constant bool:                                                   OK
    constant bool respects attributes:                               OK
    constant path:                                                   OK
    constant uri:                                                    OK
    simple set:                                                      OK
    set inherit:                                                     OK
    set scoped inherit:                                              OK
    set rec:                                                         OK
    set complex keynames:                                            OK
    set inherit direct:                                              OK
    inherit selector:                                                OK
    int list:                                                        OK
    int null list:                                                   OK
    mixed list:                                                      OK
    simple lambda:                                                   OK
    lambda or uri:                                                   OK
    lambda pattern:                                                  OK
    lambda app int:                                                  OK
    simple let:                                                      OK
    let body:                                                        OK
    nested let:                                                      OK
    let scoped inherit:                                              OK
    if:                                                              OK
    identifier special chars:                                        OK
    identifier keyword prefix:                                       OK
    simple string:                                                   OK
    string dollar:                                                   OK
    string escape:                                                   OK
    string antiquote:                                                OK
    select:                                                          OK
    select path:                                                     OK
    select keyword:                                                  OK
    fun app:                                                         OK
    indented string:                                                 OK
    indented string escape:                                          OK
    operator fun app:                                                OK
    operators:                                                       OK
    comments:                                                        OK
    select or precedence:                                            OK
    select or precedence2:                                           OK
  EvalTests
    basic sum:                                                       OK
    basic div:                                                       OK
    zero div:                                                        OK
    bit ops:                                                         OK
    basic function:                                                  OK
    set attr:                                                        OK
    function set arg:                                                OK
    function set two arg:                                            OK
    function set two arg default scope:                              OK
    function default env:                                            OK
    function definition uses environment:                            OK
    function atpattern:                                              OK
    function ellipsis:                                               OK
    function default value not in atpattern:                         OK
    function arg shadowing:                                          OK
    function recursive args:                                         OK
    function recursive sets:                                         OK
    nested with:                                                     OK
    with strictness:                                                 OK
    match failure null:                                              OK
    find file success no prefix:                                     FAIL
      Exception: NixException [Nix frame at level Error: file 'findFile.nix' was not found in the Nix search path (add it's using $NIX_PATH or -I),Nix frame at level Info: Calling "findFile" (SrcSpan {spanBegin = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 1}, spanEnd = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 72}}),Nix frame at level Info: EvaluatingExpr Scopes: [["__bitAnd","__isBool","__isString","__seq","builtins","__partition","__bitXor","__isAttrs","__bitOr","__parseDrvName","__toXML","abort","__sort","__unsafeDiscardStringContext","__hasContext","__intersectAttrs","__getAttr","__splitVersion","__isList","__tail","__isFunction","__genericClosure","__filter","__getEnv","__div","__match","__unsafeGetAttrPos","__deepSeq","__findFile","removeAttrs","__any","__langVersion","__addErrorContext","throw","__toPath","__valueSize","__getContext","__typeOf","__hasAttr","__listToAttrs","__currentSystem","__concatStringsSep","mapAttrs","__length","__sub","__elemAt","__currentTime","__mul","map","__readFile","__hashString","__fetchurl","__attrNames","__split","__foldl'","__isFloat","__nixPath","__replaceStrings","derivationStrict","__lessThan","import","__functionArgs","__readDir","__nixVersion","__pathExists","__concatLists","__false","__toFile","__concatMap","__head","__fetchTarball","__storeDir","__toJSON","baseNameOf","__null","placeholder","scopedImport","__exec","__attrValues","__genList","toString","__true","__fromJSON","__compareVersions","dirOf","__catAttrs","__elem","__add","derivation","trace","__all","__substring","isNull","__appendContext","__tryEval","__stringLength","__isInt"],["__includes"]], and 0 with-scopes (Fix (Compose (Ann {annotation = SrcSpan {spanBegin = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 1}, spanEnd = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 72}}, annotated = NBinary NApp (Fix (Compose (Ann {annotation = SrcSpan {spanBegin = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 1}, spanEnd = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 58}}, annotated = NBinary NApp (Fix (Compose (Ann {annotation = SrcSpan {spanBegin = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 1}, spanEnd = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 19}}, annotated = NSelect (Fix (Compose (Ann {annotation = SrcSpan {spanBegin = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 1}, spanEnd = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 9}}, annotated = NSym "builtins"}))) (StaticKey "findFile" :| []) Nothing}))) (Fix (Compose (Ann {annotation = SrcSpan {spanBegin = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 19}, spanEnd = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 58}}, annotated = NList [Fix (Compose (Ann {annotation = SrcSpan {spanBegin = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 20}, spanEnd = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 56}}, annotated = NSet NNonRecursive [NamedVar (StaticKey "path" :| []) (Fix (Compose (Ann {annotation = SrcSpan {spanBegin = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 27}, spanEnd = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 42}}, annotated = NStr (DoubleQuoted [Plain "./tests/files"])}))) (SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 22}),NamedVar (StaticKey "prefix" :| []) (Fix (Compose (Ann {annotation = SrcSpan {spanBegin = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 51}, spanEnd = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 53}}, annotated = NStr (DoubleQuoted [])}))) (SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 44})]}))]})))}))) (Fix (Compose (Ann {annotation = SrcSpan {spanBegin = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 58}, spanEnd = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 72}}, annotated = NStr (DoubleQuoted [Plain "findFile.nix"])})))})))]
    find file success with prefix:                                   FAIL
      Exception: NixException [Nix frame at level Error: file 'nix/findFile.nix' was not found in the Nix search path (add it's using $NIX_PATH or -I),Nix frame at level Info: Calling "findFile" (SrcSpan {spanBegin = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 1}, spanEnd = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 79}}),Nix frame at level Info: EvaluatingExpr Scopes: [["__bitAnd","__isBool","__isString","__seq","builtins","__partition","__bitXor","__isAttrs","__bitOr","__parseDrvName","__toXML","abort","__sort","__unsafeDiscardStringContext","__hasContext","__intersectAttrs","__getAttr","__splitVersion","__isList","__tail","__isFunction","__genericClosure","__filter","__getEnv","__div","__match","__unsafeGetAttrPos","__deepSeq","__findFile","removeAttrs","__any","__langVersion","__addErrorContext","throw","__toPath","__valueSize","__getContext","__typeOf","__hasAttr","__listToAttrs","__currentSystem","__concatStringsSep","mapAttrs","__length","__sub","__elemAt","__currentTime","__mul","map","__readFile","__hashString","__fetchurl","__attrNames","__split","__foldl'","__isFloat","__nixPath","__replaceStrings","derivationStrict","__lessThan","import","__functionArgs","__readDir","__nixVersion","__pathExists","__concatLists","__false","__toFile","__concatMap","__head","__fetchTarball","__storeDir","__toJSON","baseNameOf","__null","placeholder","scopedImport","__exec","__attrValues","__genList","toString","__true","__fromJSON","__compareVersions","dirOf","__catAttrs","__elem","__add","derivation","trace","__all","__substring","isNull","__appendContext","__tryEval","__stringLength","__isInt"],["__includes"]], and 0 with-scopes (Fix (Compose (Ann {annotation = SrcSpan {spanBegin = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 1}, spanEnd = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 79}}, annotated = NBinary NApp (Fix (Compose (Ann {annotation = SrcSpan {spanBegin = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 1}, spanEnd = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 61}}, annotated = NBinary NApp (Fix (Compose (Ann {annotation = SrcSpan {spanBegin = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 1}, spanEnd = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 19}}, annotated = NSelect (Fix (Compose (Ann {annotation = SrcSpan {spanBegin = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 1}, spanEnd = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 9}}, annotated = NSym "builtins"}))) (StaticKey "findFile" :| []) Nothing}))) (Fix (Compose (Ann {annotation = SrcSpan {spanBegin = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 19}, spanEnd = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 61}}, annotated = NList [Fix (Compose (Ann {annotation = SrcSpan {spanBegin = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 20}, spanEnd = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 59}}, annotated = NSet NNonRecursive [NamedVar (StaticKey "path" :| []) (Fix (Compose (Ann {annotation = SrcSpan {spanBegin = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 27}, spanEnd = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 42}}, annotated = NStr (DoubleQuoted [Plain "./tests/files"])}))) (SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 22}),NamedVar (StaticKey "prefix" :| []) (Fix (Compose (Ann {annotation = SrcSpan {spanBegin = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 51}, spanEnd = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 56}}, annotated = NStr (DoubleQuoted [Plain "nix"])}))) (SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 44})]}))]})))}))) (Fix (Compose (Ann {annotation = SrcSpan {spanBegin = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 61}, spanEnd = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 79}}, annotated = NStr (DoubleQuoted [Plain "nix/findFile.nix"])})))})))]
    find file success folder:                                        FAIL
      Exception: NixException [Nix frame at level Error: file 'files' was not found in the Nix search path (add it's using $NIX_PATH or -I),Nix frame at level Info: Calling "findFile" (SrcSpan {spanBegin = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 1}, spanEnd = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 59}}),Nix frame at level Info: EvaluatingExpr Scopes: [["__bitAnd","__isBool","__isString","__seq","builtins","__partition","__bitXor","__isAttrs","__bitOr","__parseDrvName","__toXML","abort","__sort","__unsafeDiscardStringContext","__hasContext","__intersectAttrs","__getAttr","__splitVersion","__isList","__tail","__isFunction","__genericClosure","__filter","__getEnv","__div","__match","__unsafeGetAttrPos","__deepSeq","__findFile","removeAttrs","__any","__langVersion","__addErrorContext","throw","__toPath","__valueSize","__getContext","__typeOf","__hasAttr","__listToAttrs","__currentSystem","__concatStringsSep","mapAttrs","__length","__sub","__elemAt","__currentTime","__mul","map","__readFile","__hashString","__fetchurl","__attrNames","__split","__foldl'","__isFloat","__nixPath","__replaceStrings","derivationStrict","__lessThan","import","__functionArgs","__readDir","__nixVersion","__pathExists","__concatLists","__false","__toFile","__concatMap","__head","__fetchTarball","__storeDir","__toJSON","baseNameOf","__null","placeholder","scopedImport","__exec","__attrValues","__genList","toString","__true","__fromJSON","__compareVersions","dirOf","__catAttrs","__elem","__add","derivation","trace","__all","__substring","isNull","__appendContext","__tryEval","__stringLength","__isInt"],["__includes"]], and 0 with-scopes (Fix (Compose (Ann {annotation = SrcSpan {spanBegin = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 1}, spanEnd = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 59}}, annotated = NBinary NApp (Fix (Compose (Ann {annotation = SrcSpan {spanBegin = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 1}, spanEnd = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 52}}, annotated = NBinary NApp (Fix (Compose (Ann {annotation = SrcSpan {spanBegin = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 1}, spanEnd = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 19}}, annotated = NSelect (Fix (Compose (Ann {annotation = SrcSpan {spanBegin = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 1}, spanEnd = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 9}}, annotated = NSym "builtins"}))) (StaticKey "findFile" :| []) Nothing}))) (Fix (Compose (Ann {annotation = SrcSpan {spanBegin = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 19}, spanEnd = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 52}}, annotated = NList [Fix (Compose (Ann {annotation = SrcSpan {spanBegin = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 20}, spanEnd = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 50}}, annotated = NSet NNonRecursive [NamedVar (StaticKey "path" :| []) (Fix (Compose (Ann {annotation = SrcSpan {spanBegin = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 27}, spanEnd = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 36}}, annotated = NStr (DoubleQuoted [Plain "./tests"])}))) (SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 22}),NamedVar (StaticKey "prefix" :| []) (Fix (Compose (Ann {annotation = SrcSpan {spanBegin = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 45}, spanEnd = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 47}}, annotated = NStr (DoubleQuoted [])}))) (SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 38})]}))]})))}))) (Fix (Compose (Ann {annotation = SrcSpan {spanBegin = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 52}, spanEnd = SourcePos {sourceName = "<string>", sourceLine = Pos 1, sourceColumn = Pos 59}}, annotated = NStr (DoubleQuoted [Plain "files"])})))})))]
    find file failure not found:                                     OK
    find file failure invalid arg 1:                                 OK
    find file failure invalid arg 2:                                 OK
    find file failure invalid arg no path:                           OK
    infinite recursion:                                              OK (0.05s)
    nested let:                                                      OK
    nested nested let:                                               OK
    inherit in rec set:                                              OK
    lang version:                                                    OK
    rec set attr path simpl:                                         OK
    inherit from set has no scope:                                   OK
    unsafegetattrpos1:                                               OK
    unsafegetattrpos2:                                               OK
    unsafegetattrpos3:                                               OK
    unsafegetattrpos4:                                               OK
    fixed points:                                                    OK
    fixed points and fold:                                           OK
    fixed points attrsets:                                           OK
    function equals3:                                                OK
    function equals4:                                                OK
    function equals5:                                                OK
    directory pathexists:                                            OK
    rec path attr:                                                   OK
    mapattrs builtin:                                                OK
    regression 373:                                                  OK
    empty string equal null is false:                                OK
    null equal empty string is false:                                OK
    empty string not equal null is true:                             OK
    null equal not empty string is true:                             OK
    list nested bottom diverges:                                     OK
    attrset nested bottom diverges:                                  OK
    list list nested bottom equal:                                   OK
    list attrset nested bottom equal:                                OK
    list function nested bottom equal:                               OK
    attrset list nested bottom equal:                                OK
    attrset attrset nested bottom equal:                             OK
    attrset function nested bottom equal:                            OK
    add string thunk left:                                           OK
    add string thunk right:                                          OK
    add int thunk left:                                              OK
    add int thunk right:                                             OK
    concat thunk left:                                               OK
    concat thunk rigth:                                              OK
  PrettyTests
    indented antiquotation:                                          OK
    string antiquotation:                                            OK
    function params:                                                 OK
    paths:                                                           OK
  Expr Reductions
    Non nested NSelect on set should be reduced:                     OK
    Nested NSelect on set should be reduced:                         OK
    Non nested NSelect with incorrect attrpath shouldn't be reduced: OK
    Nested NSelect with incorrect attrpath shouldn't be reduced:     OK
  Pretty/Parse Property:                                             OK
      ✓ Pretty/Parse Property passed 0 tests.
  Eval comparison tests
    ind-string-17.nix:                                               FAIL
      Exception: nix-instantiate: readCreateProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
    ind-string-16.nix:                                               FAIL
      Exception: nix-instantiate: readCreateProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
    ind-string-15.nix:                                               FAIL
      Exception: nix-instantiate: readCreateProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
    ind-string-14.nix:                                               FAIL
      Exception: nix-instantiate: readCreateProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
    ind-string-13.nix:                                               FAIL
      Exception: nix-instantiate: readCreateProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
    ind-string-12.nix:                                               FAIL
      Exception: nix-instantiate: readCreateProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
    ind-string-11.nix:                                               FAIL
      Exception: nix-instantiate: readCreateProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
    ind-string-10.nix:                                               FAIL
      Exception: nix-instantiate: readCreateProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
    ind-string-09.nix:                                               FAIL
      Exception: nix-instantiate: readCreateProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
    ind-string-08.nix:                                               FAIL
      Exception: nix-instantiate: readCreateProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
    ind-string-07.nix:                                               FAIL
      Exception: nix-instantiate: readCreateProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
    ind-string-06.nix:                                               FAIL
      Exception: nix-instantiate: readCreateProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
    ind-string-05.nix:                                               FAIL
      Exception: nix-instantiate: readCreateProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
    ind-string-04.nix:                                               FAIL
      Exception: nix-instantiate: readCreateProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
    ind-string-03.nix:                                               FAIL
      Exception: nix-instantiate: readCreateProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
    ind-string-02.nix:                                               FAIL
      Exception: nix-instantiate: readCreateProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
    ind-string-01.nix:                                               FAIL
      Exception: nix-instantiate: readCreateProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
    builtins.string.store.nix:                                       FAIL
      Exception: nix-instantiate: readCreateProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
    builtins.split-04.nix:                                           FAIL
      Exception: nix-instantiate: readCreateProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
    builtins.split-03.nix:                                           FAIL
      Exception: nix-instantiate: readCreateProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
    builtins.split-02.nix:                                           FAIL
      Exception: nix-instantiate: readCreateProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
    builtins.split-01.nix:                                           FAIL
      Exception: nix-instantiate: readCreateProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
  Nix language tests present:                                        OK
  Nix (upstream) language tests
    eval fail
      eval-fail-undeclared-arg:                                      OK
      eval-fail-to-path:                                             OK
      eval-fail-substring:                                           OK
      eval-fail-seq:                                                 OK
      eval-fail-scope-5:                                             OK
      eval-fail-remove:                                              OK
      eval-fail-path-slash:                                          OK
      eval-fail-missing-arg:                                         OK
      eval-fail-hashfile-missing:                                    OK
      eval-fail-deepseq:                                             OK
      eval-fail-blackhole:                                           OK
      eval-fail-bad-antiquote-3:                                     OK
      eval-fail-bad-antiquote-2:                                     OK
      eval-fail-bad-antiquote-1:                                     OK
      eval-fail-assert:                                              OK
      eval-fail-antiquoted-path:                                     OK
      eval-fail-abort:                                               OK
    eval okay
      eval-okay-xml:                                                 OK
      eval-okay-with:                                                OK
      eval-okay-versions:                                            OK (0.02s)
      eval-okay-tryeval:                                             OK
      eval-okay-toxml2:                                              OK
      eval-okay-toxml:                                               OK
      eval-okay-tojson:                                              OK
      eval-okay-tail-call-1:                                         OK
      eval-okay-substring:                                           OK
      eval-okay-strings-as-attrs-names:                              OK
      eval-okay-string:                                              OK
      eval-okay-splitversion:                                        OK
      eval-okay-sort:                                                OK
      eval-okay-seq:                                                 OK
      eval-okay-search-path:                                         OK
      eval-okay-scope-7:                                             OK
      eval-okay-scope-6:                                             OK
      eval-okay-scope-4:                                             OK
      eval-okay-scope-3:                                             OK
      eval-okay-scope-2:                                             OK
      eval-okay-scope-1:                                             OK
      eval-okay-replacestrings:                                      OK
      eval-okay-remove:                                              OK
      eval-okay-regex-split:                                         OK
      eval-okay-regex-match:                                         OK
      eval-okay-redefine-builtin:                                    OK
      eval-okay-readfile:                                            OK
      eval-okay-readDir:                                             OK
      eval-okay-patterns:                                            OK
      eval-okay-pathexists:                                          OK
      eval-okay-partition:                                           OK
      eval-okay-overrides:                                           OK
      eval-okay-null-dynamic-attrs:                                  OK
      eval-okay-new-let:                                             OK
      eval-okay-nested-with:                                         OK
      eval-okay-mapattrs:                                            OK
      eval-okay-map:                                                 OK
      eval-okay-logic:                                               OK
      eval-okay-listtoattrs:                                         OK
      eval-okay-list:                                                OK
      eval-okay-let:                                                 OK
      eval-okay-ind-string:                                          OK
      eval-okay-import:                                              OK
      eval-okay-if:                                                  OK
      eval-okay-hashstring:                                          OK
      eval-okay-getenv:                                              OK
      eval-okay-getattrpos-undefined:                                OK
      eval-okay-getattrpos:                                          OK
      eval-okay-functionargs:                                        OK
      eval-okay-fromjson:                                            OK
      eval-okay-float:                                               OK
      eval-okay-flatten:                                             OK (0.01s)
      eval-okay-filter:                                              OK
      eval-okay-eq-derivations:                                      FAIL
        Exception: nix-instantiate: readCreateProcessWithExitCode: runInteractiveProcess: exec: does not exist (No such file or directory)
      eval-okay-eq:                                                  OK
      eval-okay-empty-args:                                          OK
      eval-okay-elem:                                                OK
      eval-okay-dynamic-attrs-bare:                                  OK
      eval-okay-dynamic-attrs-2:                                     OK
      eval-okay-dynamic-attrs:                                       OK
      eval-okay-delayed-with-inherit:                                OK
      eval-okay-delayed-with:                                        OK
      eval-okay-deepseq:                                             OK
      eval-okay-curpos:                                              OK
      eval-okay-context:                                             FAIL
        Exception: nix-store: readCreateProcessWithExitCode: runInteractiveProcess: exec: does not exist (No such file or directory)
      eval-okay-concatstringssep:                                    OK
      eval-okay-concatmap:                                           OK
      eval-okay-concat:                                              OK
      eval-okay-comments:                                            OK
      eval-okay-closure:                                             OK (0.13s)
      eval-okay-catattrs:                                            OK
      eval-okay-callable-attrs:                                      OK
      eval-okay-builtins-add:                                        OK
      eval-okay-builtins:                                            OK
      eval-okay-backslash-newline-2:                                 OK
      eval-okay-backslash-newline-1:                                 OK
      eval-okay-autoargs:                                            OK
      eval-okay-attrs5:                                              OK
      eval-okay-attrs4:                                              OK
      eval-okay-attrs3:                                              OK
      eval-okay-attrs2:                                              OK
      eval-okay-attrs:                                               OK
      eval-okay-attrnames:                                           OK
      eval-okay-arithmetic:                                          OK (0.01s)
      eval-okay-any-all:                                             OK
    parse fail
      parse-fail-undef-var-2:                                        OK
      parse-fail-undef-var:                                          OK
      parse-fail-uft8:                                               OK
      parse-fail-regression-20060610:                                OK
      parse-fail-patterns-1:                                         OK
      parse-fail-mixed-nested-attrs2:                                OK
      parse-fail-mixed-nested-attrs1:                                OK
      parse-fail-dup-formals:                                        OK
      parse-fail-dup-attrs-7:                                        OK
      parse-fail-dup-attrs-4:                                        OK
      parse-fail-dup-attrs-3:                                        OK
      parse-fail-dup-attrs-2:                                        OK
      parse-fail-dup-attrs-1:                                        OK
    parse okay
      parse-okay-url:                                                OK
      parse-okay-subversion:                                         OK
      parse-okay-regression-751:                                     OK
      parse-okay-regression-20041027:                                OK
      parse-okay-mixed-nested-attrs-3:                               OK
      parse-okay-mixed-nested-attrs-2:                               OK
      parse-okay-mixed-nested-attrs-1:                               OK
      parse-okay-dup-attrs-6:                                        OK
      parse-okay-dup-attrs-5:                                        OK
      parse-okay-crlf:                                               OK
      parse-okay-1:                                                  OK

27 out of 265 tests failed (0.55s)
Test suite hnix-tests: FAIL
Test suite logged to: dist/test/hnix-0.8.0-hnix-tests.log
0 of 1 test suites (0 of 1 test cases) passed.
builder for '/nix/store/y29x5khwlm8cp0rq9rd38518gx9hjsql-hnix-0.8.0.drv' failed with exit code 1
error: build of '/nix/store/y29x5khwlm8cp0rq9rd38518gx9hjsql-hnix-0.8.0.drv' failed
```
</details>

**Comment by sorki** [comment]
Good idea for now though it would be better to only depend on `corepkgs` part of the submodule or just `nix` package as the submodule is now responsible for ~600 lines of `extra-source-files` in `hnix.cabal`.

From one my notes, I think it also looks into wrong path now while trying to find `corepkgs` but I don't rember exactly. I was also able to point `hnix` to systems `nix` `corepkgs` as well via `NIX_DATA_DIR`

```
NIX_DATA_DIR=/home/srk/git/hnix/data/nix NIX_PATH=nixpkgs=/home/srk/git/nixpkgs hnix --eval --strict ./go.nix

 ^^ looks for .../data/nix/nix/corepkgs (fixed by hand)
```

**Comment by sjakobi** [comment]
> Good idea for now though it would be better to only depend on `corepkgs` part of the submodule or just `nix` package as the submodule is now responsible for ~600 lines of `extra-source-files` in `hnix.cabal`.

I suspect that most of these `extra-source-files` are unnecessary, since we only seem to use the test files in `data/nix/tests/lang`. We could also use globs to avoid listing each individual file.

I could give that a spin if you think it's worthwile.

**Comment by Anton-Latukha** [comment]
I belabor what I know from author words and from https://github.com/haskell-nix/hnix/issues/641#issue-643281649, `nix-instantiate` is the main sticking point. `nix-instantiate` uses who knows what files in Nix submodule, so that is why all those files needed to be mentioned in Cabal description and dragged around with our project.
When we free from `nix-instantiate` - we can slash the usage of the files in the submodule, and then even strip it to the list of what is really used by the HNix itself, as you say - `data/nix/tests/lang` mostly.

**Comment by Anton-Latukha** [comment]
There are tests that still need to be sorted out before updating submodule.

**Comment by soulomoon** [comment]
> There are tests that still need to be sorted out before updating submodule.

I am testing out the current branch with submodule updated to nix 2.4. Will collect some issues here.
*  Eval comparison tests
   *  ~~ind-string-05.nix ind-string-14.nix, escape problem, #1000~~
*  Nix (upstream) language tests 
   *  eval-okay-tojson, seems to be __toString problem, #198 
   *  ~~eval-okay-ind-string, escape problem, #1000~~
   *  ~~eval-okay-fromjson-escapes, escape problem, #1000~~
   *  eval-okay-path-antiquotation, path antiquotation, no opened issue is found yet
   * eval-okay-floor-ceil, builtins.floor and builtins.ceil are missing, no opened issue is found yet
   * val-okay-attrs6, to be filled
   * eval-okay-search-path, to be filled
   * eval-okay-getattrpos-functionargs, to be filled

**Comment by Anton-Latukha** [comment]
There is: 
https://github.com/haskell-nix/hnix/blob/64ba3c78c6197b81a3b70a8978cffcfe42b35b7d/tests/NixLanguageTests.hs#L54-L64

I even wanted to put escaping tests there until the solution is found.

**Comment by soulomoon** [comment]
> There is:
> 
> https://github.com/haskell-nix/hnix/blob/64ba3c78c6197b81a3b70a8978cffcfe42b35b7d/tests/NixLanguageTests.hs#L54-L64
> 
> I even wanted to put escaping tests there until the solution is found.

I will try to tackle some of them and see what I can do to help for the `newFailingTests`.

I believe most of the escaping tests is solved in #1019，please review it when you are free.
